manager/defaultmanager/copy.go

```diff
@@ -0,0 +1,165 @@
+package defaultmanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	load "github.com/uor-framework/uor-client-go/config"
+	"github.com/uor-framework/uor-client-go/model"
+	"github.com/uor-framework/uor-client-go/ocimanifest"
+	"github.com/uor-framework/uor-client-go/registryclient"
+	"github.com/uor-framework/uor-client-go/util/workspace"
+	"oras.land/oras-go/v2/content/memory"
+)
+
+func (d DefaultManager) Update(ctx context.Context, space workspace.Workspace, src string, dest string, add bool, remove bool, client registryclient.Client) (string, error) {
+
+	// Pull the dataset config from the target collection
+	_, manBytes, err := client.GetManifest(ctx, src)
+	if err != nil {
+		return "", err
+	}
+
+	var manifest ocispec.Manifest
+	if err := json.NewDecoder(manBytes).Decode(&manifest); err != nil {
+		return "", err
+	}
+	manconfig, err := client.GetContent(ctx, src, manifest.Config)
+	if err != nil {
+		return "", err
+	}
+
+	config, err := load.LoadDataSetConfig(manconfig)
+	if err != nil {
+		return "", err
+	}
+
+	if remove {
+		pruneDescs, err := d.pullCollection(ctx, src, memory.New(), client)
```

Comment:
Depending on what files are being deleted, this action could run the system out of memory since this is storing all of the files as byte slices.